src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/misc/StrangeNetherGoo.java

```diff
@@ -56,4 +66,26 @@ public ItemUseHandler getItemHandler() {
         };
     }
 
+    private EntityInteractHandler onRightClickEntity() {
+        return (e, item, hand) -> {
+            if ((e.getRightClicked() instanceof Sheep) && (e.getRightClicked() instanceof LivingEntity)) {
+                Sheep s = (Sheep) e.getRightClicked();
+
+                if (s.getCustomName() != null) {
+                    return;
```

Comment:
You should probably cancel the event here too to prevent them from dying the sheep.